gateway-provider-security-jwt/src/test/java/org/apache/knox/gateway/provider/federation/jwt/filter/SignatureVerificationCacheTest.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.provider.federation.jwt.filter;
+
+import com.nimbusds.jwt.SignedJWT;
+import org.apache.knox.gateway.provider.federation.TestFilterConfig;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class SignatureVerificationCacheTest {
+
+    private static RSAPrivateKey privateKey;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+        kpg.initialize(2048);
+        KeyPair KPair = kpg.generateKeyPair();
+        privateKey = (RSAPrivateKey) KPair.getPrivate();
+    }
+
+    /**
+     * Verify that the default SignatureVerificationCache instance is returned when no topology is specified.
+     */
+    @Test
+    public void testSignatureVerificationCacheWithoutTopology() throws Exception {
+        SignatureVerificationCache cache = SignatureVerificationCache.getInstance(null, new TestFilterConfig());
+        assertNotNull(cache);
+        SignatureVerificationCache defaultCache =
+            SignatureVerificationCache.getInstance(SignatureVerificationCache.DEFAULT_CACHE_ID, new TestFilterConfig());
+        assertNotNull(defaultCache);
+        assertEquals("Expected the default cache when no topology is specified.", defaultCache, cache);
+    }
+
+    /**
+     * Verify that the topology-specific SignatureVerificationCache instance is returned when a topology is specified.
+     */
+    @Test
+    public void testSignatureVerificationCacheForTopology() throws Exception {
+        final String topologyName = "test-topology-explicit";
+        final Properties filterProps = new Properties();
+        filterProps.setProperty(TestFilterConfig.TOPOLOGY_NAME_PROP, topologyName);
+        final TestFilterConfig filterConfig = new TestFilterConfig(filterProps);
+
+        SignatureVerificationCache ref1 = SignatureVerificationCache.getInstance(topologyName, filterConfig);
+        assertNotNull(ref1);
+
+        SignatureVerificationCache ref2 = SignatureVerificationCache.getInstance(topologyName, filterConfig);
+        assertNotNull(ref2);
+        assertEquals("Expected the same cache when the topology is explicitly specified.", ref1, ref2);
```

Comment:
Maybe adding another ref with a different topology name and assert they are not the same?